sentry-rails/lib/sentry/rails/tracing/abstract_subscriber.rb

```diff
@@ -41,14 +41,18 @@ def record_on_current_span(duration:, **options)
             return unless options[:start_timestamp]
 
             scope = Sentry.get_current_scope
-            transaction = scope.get_transaction
-            return unless transaction && transaction.sampled
+            current_span = scope.get_span
+            return unless current_span && current_span.sampled
+
+            span = current_span.start_child(**options)
+            Sentry.get_current_scope.set_span(span)
```

Comment:
I think this and the `get_current_scope` are unnecessary. Aren't they the same scope as the `scope` local?